src/binder/expression_binder.cpp

```diff
@@ -143,23 +143,38 @@ shared_ptr<Expression> ExpressionBinder::bindNodePropertyExpression(
     shared_ptr<Expression> node, const string& propertyName) {
     auto catalogContent = binder->catalog.getReadOnlyVersion();
     auto nodeExpression = static_pointer_cast<NodeExpression>(node);
-    if (nodeExpression->getNumTableIDs() > 1) {
-        throw BinderException("Cannot bind property for multi-labeled node " + node->getRawName());
+    vector<Property> properties;
+    for (auto tableID : nodeExpression->getTableIDs()) {
+        if (!catalogContent->containNodeProperty(tableID, propertyName)) {
+            continue;
+        }
+        properties.push_back(catalogContent->getNodeProperty(tableID, propertyName));
     }
-    if (catalogContent->containNodeProperty(nodeExpression->getTableID(), propertyName)) {
-        auto& property =
-            catalogContent->getNodeProperty(nodeExpression->getTableID(), propertyName);
-        return bindNodePropertyExpression(node, property);
-    } else {
-        throw BinderException("Node " + nodeExpression->getRawName() + " does not have property " +
-                              propertyName + ".");
+    if (properties.empty()) {
+        throw BinderException(
+            "Cannot find property " + propertyName + " under node " + node->getRawName());
+    }
+    // Validate data type.
+    auto dataType = properties[0].dataType;
+    for (auto& property : properties) {
+        if (property.dataType != dataType) {
+            throw BinderException("Cannot resolve data type for property " + propertyName +
+                                  " under node " + node->getRawName());
+        }
     }
+    return bindNodePropertyExpression(node, std::move(properties));
 }
 
 shared_ptr<Expression> ExpressionBinder::bindNodePropertyExpression(
-    shared_ptr<Expression> node, const Property& property) {
-    return make_shared<PropertyExpression>(
-        property.dataType, property.name, property.propertyID, move(node));
+    shared_ptr<Expression> node, const vector<Property>& properties) {
+    assert(!properties.empty());
+    auto propertyAnchor = properties[0];
+    unordered_map<table_id_t, property_id_t> propertyIDPerTable;
+    for (auto& property : properties) {
```

Comment:
I would put an assert here for the equality of name and data type of property with propertyAnchor, so the logic of this function is more self-explained.